### PR TITLE
#164859749 User reset password via email

### PR DESCRIPTION
**Pull request by jmusila**
## What does this PR do?
- Creates user reset password endpoint

## Description of Task to be completed?
- [x] Create user reset password endpoint

## How should this be manually tested?
- Clone the repo to your local machine from here

- cd into the ah-django directory and checkout to the branch `ft-reset-password-viaemail-164859749`

- `cd ah-django`

- git checkout `ft-reset-password-viaemail-164859749`

- Create a virtual environment and activate it:

- `python3 -m virtualenv venv && source venv/bin/activate`

- If you need to install virtualenv first:

- `pip install virtualenv`

- Install the project dependencies

- `pip install -r requirements.txt`

- Make database migrations and start the django server

- `python3 manage.py makemigrations`

- `python3 manage.py migrate`

- `python3 manage.py runserver`

- signup as a new user using a valid email

- try changing passing using the signed up email as the input

- check your inbox to find the send link and click on it to reset the password

- for `receiving a reset link` provide the data to postman in this format:
  ```  
     {
	 "email": "test.user@gmail.com"
     }
  ```
- for `resetting the password` provide the data to postman in this format:
  ```
     {
       "password": "mYp@55ord"
       "password": "mYp@55ord"
     }
  ```
- request type `POST`
## Any background context you want to provide?
- Initially, the application had no reset password endpoint

## What are the relevant pivotal tracker stories?
#164859749

## Screenshots:
Insomnia – http:--127 0 0 1:8000-api-v1-users-forgot_password_161
Selection_160
Insomnia – http:--127 0 0 1:8000-api-v1-users-forgot_password_166
Selection_164

### CODE REVIEW

**Review by philophilo — CHANGES_REQUESTED** [review]
All these changes are not necessary, we are dealing with backend alone.

File: authors/apps/authentication/urls.py [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 from .views import (
-    LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView
+    LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView,
+    ForgotPasswordAPIview, ResetPasswordAPIView
 )
 app_name = 'authentication'
 
+
 urlpatterns = [
-    url(r'^user/?$', UserRetrieveUpdateAPIView.as_view()),
```
Comment (philophilo) [inline authors/apps/authentication/urls.py]: Any reason we are getting rid of this endpoint

Comment (jmusila) [reply authors/apps/authentication/urls.py]: Yeah I think I will remove that

Comment (philophilo) [reply authors/apps/authentication/urls.py]: @jonathanmusila the reason

Comment (jmusila) [reply authors/apps/authentication/urls.py]: I think I got you now, no reason to get rid of it, I think I removed that by mistake, will correct that

File: authors/apps/authentication/tests/base_test.py [hunk 2]
```diff
@@ -33,10 +34,38 @@ def setUp(self):
...
+        }
+
+        self.reset_password_data = {
+            "password": "Itr!3d21",
+            "confirm_password": "Itr!3d21"
+        }
+
+        self.reset_password = {
+            "email": "",
+        }
```
Comment (philophilo) [inline authors/apps/authentication/tests/base_test.py:53]: use descriptive names

File: authors/apps/authentication/tests/base_test.py [hunk 3]
```diff
@@ -33,10 +34,38 @@ def setUp(self):
...
+
+        self.reset_password1 = {
+            "email": "cow@mammals.milk",
+        }
```
Comment (philophilo) [inline authors/apps/authentication/tests/base_test.py:57]: here too

File: authors/apps/authentication/tests/base_test.py (hunk 2 above)
Comment (jmusila) [reply authors/apps/authentication/tests/base_test.py:53]: Done

File: authors/apps/authentication/tests/test_auser_account.py [hunk 4]
```diff
@@ -11,20 +11,6 @@ class TestUser(BaseTestCase):
         Holds user account tests
     """
 
-    def test_new_user_can_register(self):
-        """
-            A user should be informed of a verification email sent to
```
Comment (philophilo) [inline authors/apps/authentication/tests/test_auser_account.py:16]: why do we need to do away with this test

Comment (jmusila) [reply authors/apps/authentication/tests/test_auser_account.py:16]: The test is still there I just transferred it to file `test_amail_send` I thought my test was depending on it

Comment (philophilo) [reply authors/apps/authentication/tests/test_auser_account.py:16]: Great...

**Review by philophilo — APPROVED** [review]

**Pull request merged**